File: Algorithms/Sorting/BubbleSort/BubbleSort.java
package Algorithms.Sorting.BubbleSort;

public class BubbleSort {
    
    static void bubbleSort(int[] arr, int n) {
        for (int i = n-1; i >= 0; i--) {     // ---- n
            for (int j = 0; j <= i - 1; j++) {     // ----- n
                if (arr[j] > arr[j+1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }

        // Time complexity - n * n = n^2
        // O(N^2)
        System.out.println("After bubble sort: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String args[]) {
        int arr[] = {13, 46, 24, 52, 20, 9};
        int n = arr.length;
        System.out.println("Before Using Bubble Sort: ");
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        bubbleSort(arr, n);
    }
}

File: Algorithms/Sorting/BubbleSort/BubbleSort.py
from typing import List

# Time complextity: O(n^2)
# Space complexity: O(1)
class BubbleSort:
    def bubblesort(self, arr: List[int]) -> List[int]:
        n = len(arr)

        for i in range(n):
            for j in range(0, n-i-1):
                if arr[j] > arr[j+1]:
                    arr[j], arr[j+1] = arr[j+1], arr[j]

        return arr
    
def main():
    sol = BubbleSort()
    arr = [64, 34, 25, 12, 22, 11, 90]
    sorted_arr = sol.bubblesort(arr)
    print("Sorted array is:", sorted_arr)

if __name__ == "__main__":
    main()